### ISSUE TITLE
nf4_gemm_transpose PTX kernel fails on Blackwell GB10 (sm_121)

### ISSUE THREAD

**Issue by noahgift** [paiml/trueno#184]
## Bug Report

**Component**: trueno-gpu PTX codegen (`nf4_gemm_transpose`)
**GPU**: NVIDIA GB10 (Blackwell, compute 12.1, sm_121)
**Driver**: 580.126.09, CUDA 13.0
**Platform**: aarch64 (NVIDIA Grace Blackwell GX10)

### Problem

The `nf4_gemm_transpose` kernel fails to load via both `cuModuleLoadDataEx` (CUDA_ERROR_INVALID_IMAGE, code 300) and `cuModuleLoadData` (CUDA_ERROR_ILLEGAL_ADDRESS, code 700) on the Blackwell GB10.

Other NF4 kernels (`nf4_gemm_fused`, `residual_add`, `softmax`, `swiglu`, `cross_entropy`, `adamw_step`, etc.) all compile and run correctly via the `cuModuleLoadData` fallback path.

### Impact

- QLoRA backward pass fails at layer 35 → poisons entire CUDA context
- All subsequent GPU operations fail with ILLEGAL_ADDRESS
- Training falls back to CPU (unusable for 4B param model on ARM)
- Blocks all entrenar CUDA training on Blackwell/GB10 hardware

### Reproduction

```bash
# On GB10 (sm_121):
entrenar train ssc-chat-qwen3-4b-qlora.yaml --verbose
# Forward pass succeeds (all kernels JIT-compile via cuModuleLoadData fallback)
# Backward pass layer 35: nf4_gemm_transpose_511_9728_2560 fails
# CUDA context poisoned, all subsequent steps fail
```

### Error log

```
[PTX-JIT] Both attempts failed for .visible .entry nf4_gemm_transpose(
[backward_nf4] Layer 35 FAILED: KernelError("Failed to compile nf4_gemm_transpose_511_9728_2560: 
  ModuleLoad(\"CUDA module loading failed: try1=300 try2=700 (JIT target: sm_121)\")") 
  (seq_len=511, hidden=2560)
```

### Analysis

- PTX targets `.target sm_70` (also tried sm_90) — both fail
- The `nf4_gemm_transpose` and `nf4_gemm_fused` PTX are structurally similar (same NF4 LUT init, same shared memory pattern)
- Key difference: transpose kernel uses different indexing (`16x16` tiles vs `32x32`, `%r5` vs `%r4` for bounds)
- The `cuModuleLoadData` fallback returns 700 (ILLEGAL_ADDRESS) — unusual for a module load call, suggests driver-level PTX validation failure on Blackwell
- All other kernels (including `nf4_gemm_fused` with identical shared memory LUT pattern) succeed

### Proposed fix

Replace the custom `nf4_gemm_transpose` PTX with a cuBLAS-based implementation for the backward pass GEMM, or diagnose the specific PTX instruction that Blackwell's JIT rejects.

### Environment

- trueno-gpu 0.4.31 (local path dep)
- entrenar 0.7.5
- Rust 1.94.0 (aarch64-unknown-linux-gnu)
- NVIDIA GB10, Driver 580.126.09, CUDA 13.0
